Mage/Archivist/Archivist.cs

```diff
@@ -181,22 +187,27 @@ public class Archivist : EventEmitter<JObject> {
 				return;
 			}
 
-			JArray topicValues = (JArray)result;
-			foreach (JObject topicValue in topicValues) {
-				// Determine value cacheKeyName
-				string valueTopic = topicValue["key"]["topic"].ToString();
-				JObject valueIndex = (JObject)topicValue["key"]["index"];
-				string cacheKeyName = getCacheKey(valueTopic, valueIndex);
-
-				// Create vault value
-				VaultValue newValue = new VaultValue(topicValue);
-
-				// Add value to cache
-				_cache.Add (cacheKeyName, newValue);
-
-				// Add value to response
-				int responseKey = realQueryKeys[cacheKeyName];
-				responseArray[responseKey].Replace((JToken)newValue.data);
+			try {
```

Comment:
It seems virtually every line below could blow up. So I think this is somewhat of a shortcut to keep it safe. Assume sane object structure, catch it when it's not sane.